### DISCUSSION TITLE
Support for multiple first level toc files in same directory

### DISCUSSION THREAD

**Discussion by chri5topherus** [dotnet/docfx#7862] [Q&A] [answered]
We try to build multiple documentations for one project with gitlab ci and docfx. Each documentation is started from its own docfx.json file: 

- docfx-webserver1.json
- docfx-webserver2.json
- docfx-webserver3.json

The problem starts with the toc files. If the files are located in sub directories, the first level of the menu is missing. If the toc files are renamed, the following error occurs: 

- Please add `YamlMime` as the first line of file

After adding `### YamlMime:ManagedReference` this error occurs: 

- Unable to load file 'toc-webserver1.yml' via processor 'ManagedReferenceDocumentProcessor'

Any idea on how to achieve this?
Thanks a lot.

**Comment by KalleOlaviNiemitalo** [comment]
> If the files are located in sub directories, the first level of the menu is missing.

How to reproduce this?

**Reply by chri5topherus** [reply]
Thanks for the response. A sample project attached where the starting toc is located in `webserver1`as set in `docfx.json`line 8. 

By building and serving the site locally. The header menu is missing with `Articles`. If the toc is moved to the root of this project everything is working fine. 

Do I have to define the toc path manually? 

[Archive.zip](https://github.com/dotnet/docfx/files/7939718/Archive.zip)

**Comment by KalleOlaviNiemitalo** [comment, marked as answer]
If you want to use `webserver1/toc.yml` as the top-level TOC, you can do it like this:

```JSON
{
  "build": {
    "content": [
      {
        "files": [
          "articles/**.md",
          "articles/**/toc.yml",
          "*.md"
        ]
      },
      {
        "files": [
          "toc.yml"
        ],
        "src": "webserver1"
      }
    ],
    "resource": [
      {
        "files": [
          "images/**"
        ]
      }
    ],
    "overwrite": [
      {
        "exclude": [
          "obj/**",
          "_site/**"
        ]
      }
    ],
    "dest": "_site",
    "globalMetadataFiles": [],
    "fileMetadataFiles": [],
    "template": [
      "default"
    ],
    "postProcessors": [],
    "markdownEngineName": "markdig",
    "noLangKeyword": false,
    "keepFileLink": false,
    "cleanupCacheHistory": false,
    "disableGitFeatures": false
  }
}
```

The `"src": "webserver1"` setting makes DocFX read `toc.yml` from the `webserver1` directory in the source tree. Because the "dest" property is not set, the generated `toc.html` still goes to the root directory of the site. The "src" and "dest" properties of the file mapping object are documented here: <https://github.com/dotnet/docfx/blob/3acf0c1f6a4dc91363cc927bfc5fd4c61dd1c7f2/Documentation/tutorial/docfx.exe_user_manual.md#L565-L566>

**Reply by chri5topherus** [reply]
Hi, 
thanks a lot. It solved my problem using multiple TOC files.
